ampe.c

```diff
@@ -429,8 +430,9 @@ static int check_frame_protection(struct candidate *cand, struct ieee80211_mgmt_
     ampe_ie_len = len -
                 (elems->mic + elems->mic_len - (unsigned char *)mgmt);
 
-    /* expect at least MGTK + RSC + expiry */
-    if (ampe_ie_len < 2 + sizeof(struct ampe_ie) + 16 + 8 + 4) {
+    /* expect at least MGTK + RSC + expiry for open/confirm */
+    if (ftype != PLINK_CLOSE &&
```

Comment:
AMPE IE is available in the mesh peering close frame defined in section 8.5.16.4.2 IEEE802.11-2012.